tests/FSharp.Compiler.ComponentTests/OCamlCompat/OCamlCompat.fs

```diff
@@ -1,33 +1,146 @@
-    [<Theory; Directory(__SOURCE_DIRECTORY__ + "/../resources/tests/OCamlCompat", Includes=[|"IndentOff02.fs"|])>]
-    let ``OCamlCompat - IndentOff02.fs - --warnaserror --mlcompatibility`` compilation =
-        compilation
+    [<Fact>]  //	SOURCE=E_IndentOff01.fs  COMPILE_ONLY=1 SCFLAGS="--warnaserror --test:ErrorRanges"				# E_IndentOff01.fs
+    let ``OCamlCompat - --warnaserror --test:ErrorRanges E_IndentOff01.fs`` () =
+        FSharp (loadSourceFromFile (Path.Combine(__SOURCE_DIRECTORY__, "E_IndentOff01.fs")))
+        |> asFsx
+        |> withOptions ["--warnaserror"; "--test:ErrorRanges"]
+        |> compile
+        |> shouldFail
+        |> withErrorCode 0062
+        |> withDiagnosticMessageMatches "This construct is for ML compatibility\. Consider using a file with extension '\.ml' or '\.mli' instead\. You can disable this warning by using '--mlcompatibility' or '--nowarn:62'\."
+
+
+    [<Fact>]
+    let ``OCamlCompat - --warnaserror"; "--mlcompatibility IndentOff02.fs`` () =
+        FSharp (loadSourceFromFile (Path.Combine(__SOURCE_DIRECTORY__, "IndentOff02.fs")))
         |> asFsx
         |> withOptions ["--warnaserror"; "--mlcompatibility"]
         |> typecheck
         |> shouldSucceed
-        |> ignore
 
-    // This test was automatically generated (moved from FSharpQA suite - OCamlCompat)
+
     //<Expects status="warning" span="(4,1-4,14)" id="FS0062">This construct is for ML compatibility\. Consider using a file with extension '\.ml' or '\.mli' instead\. You can disable this warning by using '--mlcompatibility' or '--nowarn:62'\.$</Expects>
-    [<Theory; Directory(__SOURCE_DIRECTORY__ + "/../resources/tests/OCamlCompat", Includes=[|"W_IndentOff03.fs"|])>]
-    let ``OCamlCompat - W_IndentOff03.fs - --test:ErrorRanges`` compilation =
-        compilation
+    [<Fact>]
+    let ``OCamlCompat - --test:ErrorRanges W_IndentOff03.fs`` () =
+        FSharp (loadSourceFromFile (Path.Combine(__SOURCE_DIRECTORY__, "W_IndentOff03.fs")))
         |> asFsx
         |> withOptions ["--test:ErrorRanges"]
         |> compile
         |> shouldFail
         |> withWarningCode 0062
-        |> withDiagnosticMessageMatches "This construct is for ML compatibility\. Consider using a file with extension '\.ml' or '\.mli' instead\. You can disable this warning by using '--mlcompatibility' or '--nowarn:62'\.$"
-        |> ignore
+        |> withDiagnosticMessageMatches "This construct is for ML compatibility\. Consider using a file with extension '\.ml' or '\.mli' instead\. You can disable this warning by using '--mlcompatibility' or '--nowarn:62'\."
+
+
+    [<Fact>]    //NoMT	SOURCE=IndentOff04.fsx   COMPILE_ONLY=1 SCFLAGS="--warnaserror --mlcompatibility" FSIMODE=PIPE					# IndentOff04.fsx
+    let ``OCamlCompat - --warnaserror --mlcompatibility IndentOff04.fsx`` () =
+        FSharp (loadSourceFromFile (Path.Combine(__SOURCE_DIRECTORY__, "IndentOff04.fsx")))
```

Comment:
Curious, why not use the `[<Directory>]` attribute? It will load read the file automatically, if `Includes` parameter is specified. If not specified, all files will be loaded.

Reply:
When I first did a test like this reading from a file I didn't know about DirectoryAttribute, when I found out about the attribute, I thought meh!!! and didn't bother to change the code.  While driving home I decided I would like the output files to be built in the artifacts directory which would be improved by the use of the directory attribute.

I think I will implement a test improvement to do that and at the same time update all of these tests to use the directory attribute.  It should be noted that the globalization tests ported a few weeks ago don't use the directoryattribute either.